docs/adr/0004-runtime-governance.md

```diff
@@ -0,0 +1,225 @@
+  runtime message subject to the `runtime_update_delay` registry consensus
+  parameter. The runtime owner (indicated by `entity_id`) is not able to perform
+  any updates after the initial registration and such attempts must return
+  `ErrForbidden`.
+
+  The runtime itself is the one that needs to provide the required stake in
+  escrow in order to avoid the runtime from being suspended. This assumes that
+  runtimes can have accounts in the staking module as specified by [ADR 0003].
+  Note that anyone can delegate the required stake to a runtime in order to
+  enable its operation.
+
+- **Consensus layer governance (`GovernanceConsensus`).** In this case only the
+  consensus layer itself can update the runtime descriptor either through a
+  network upgrade or via a consensus layer governance mechanism not specified by
+  this proposal.
+
+  Runtimes using this governance model are never suspended and do not need to
+  provide stake in escrow.
+
+  Runtimes using this governance model cannot be registered via regular registry
+  method calls (doing so must return `ErrForbidden`). Instead such a runtime can
+  only be registered at genesis, through a network upgrade or via a consensus
+  layer governance mechanism not specified by this proposal.
+
+### State
+
+This proposal introduces/updates the following consensus state in the registry
+module:
+
+#### Stored Runtime Descriptors
+
+Since the runtime descriptors can now be updated by actors other than the
+initial registering entity, it does not make sense to store signed runtime
+descriptors. The value of storage key prefixed with `0x13` which previously
+contained signed runtime descriptors is modified to store plain runtime
+descriptors.
+
+#### Pending Runtime Updates
+
+Key format:
+
+| Field         | Type                           | Content                    |
+|:--------------|:------------------------------:|:--------------------------:|
+| prefix        | `0x1A` (1 byte)                | state key prefix           |
+| H(runtime-id) | H(common.Namespace) (32 bytes) | runtime identifier         |
+
+Value:
+
+```golang
+type PendingRuntimeUpdate struct {
+    Epoch      epochtime.EpochTime `json:"epoch"`
+    Descriptor registry.Runtime    `json:"descriptor"`
+}
+```
+
+### Transaction Methods
+
+This proposal updates the following transaction methods in the registry module:
+
+#### Register Runtime
+
+Runtime registration enables a new runtime to be created or an existing runtime
+to be updated (in case the governance model allows it).
+
+**Method name:**
+
+```
+registry.RegisterRuntime
+```
+
+The body of a register runtime transaction must be a `Runtime` descriptor.
+The signer of the transaction must be the owning entity key.
+
+Registering a runtime may require sufficient stake in either the owning entity's
+(when entity governance is used) or the runtime's (when runtime governance is
+used) escrow account.
```

Comment:
what is the workflow for creating a runtime with runtime governance? do you create it that way, and it's immediately suspended and then you do something? or maybe you create it under entity governance, and you transfer some tokens and then you update it to runtime governance?

Reply:
You can escrow tokens to the runtime account before registering the runtime itself. And then you can register it with runtime governance directly.